apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -0,0 +1,456 @@
+                        accountIndex[accountKey] = {
+                            dv: dv,
+                            sender: sender,
+                            context: { id: billingContext.id },
+                            number: accountNumber,
+                            importId: accountKey,
+                            globalId: accountKey,
+                            unitName,
+                            unitType,
+                            property: propertyIndex[propertyKey],
+                            raw: { dv: 1 },
+                            meta: { dv: 1 },
+                        }
+                    }
+
+                    const receipt = { category, period, property, account, services, recipient: { tin, iec, bic, bankAccount } }
+                    const receiptKey = getBillingReceiptKey(receipt)
+
+                    if (!receiptIndex[receiptKey]) {
+                        receiptIndex[receiptKey] = {
+                            dv: dv,
+                            sender: sender,
+                            context: { id: billingContextId },
+                            account: accountIndex[accountKey],
+                            property: propertyIndex[propertyKey],
+                            period: period,
+                            importId: importId,
+                            category: { id: category.id },
+                            toPay: toPay,
+                            services: services,
+                            toPayDetails: toPayDetails,
+                            tin,
+                            iec,
+                            bic,
+                            bankAccount,
+                            raw: { ...{ dv: 1 }, ...raw },
+                        }
+                    }
+                }
+
+                // Step 2:
+                // Sync billing properties
+                const syncedProperties = await syncBillingProperties(context, Object.values(propertyIndex), { billingContextId })
+
+                // Step 3:
+                // Sync Billing Accounts
+                const syncedAccounts = await syncBillingAccounts(context, Object.values(accountIndex), { properties: syncedProperties, billingContextId })
+
+                // Step 4:
+                // Sync billing receipts
+                const { createdReceipts, updatedReceipts, notChangedReceipts } = await syncBillingReceipts(context, Object.values(receiptIndex), { accounts: syncedAccounts, properties: syncedProperties, billingContextId })
+
+                // To form a result with partial success a list of promises needs to be returned from function
+                // NOTE: getById is a hack that help to resolve complex fields
+                const resultData = [...createdReceipts, ...updatedReceipts].map(item => getById('BillingReceipt', item.id))
+                const resultErrors = partialErrors.map(err => (new Promise(() => { throw err })))
```

Comment:
it looks like you change the order of rows if case of error. is it ok?

Reply:
No, this is not ok. Gonna move out from graphQL errors

Reply:
Fixed this by adding a field inputIndex to grahphQL error body

Reply:
Done.

Reply:
Now there are a param called inputIndex, which describes the path